[33747]Back button does not work when coming from Boards

bsatarnejad opened the pull request:
Make the back button work in boards after opening a WP in split screen.

https://community.openproject.com/projects/openproject/work_packages/33747/activity

bsatarnejad committed:
make the back button work in boards

HDinger reviewed (commented):
Hi @bsatarnejad 

The bug itself is fixed, however due to the changes another bug occurred: When I start on the BCF module, go the full screen and back again, the tab has changed:
![tab_is_lost](https://user-images.githubusercontent.com/7457313/86470228-48c7df80-bd3b-11ea-8a61-7e8b12baa82f.gif)
 When going back, I want to see the same tab as before (and not the activity tab).

Further, could you please write tests for these cases so that they don't happen in the future again? So one test for the actual bug behaviour and one for keeping the tab when coming from the BCF module. For the BCF test you can have a look here: https://github.com/opf/openproject/blob/dev/modules/bim/spec/features/bim_navigation_spec.rb#L124

bsatarnejad committed:
writing a test for navigating between different tabs

bsatarnejad committed:
navigating between different views with different tabs in boards

bsatarnejad committed:
ignoring show and activity and only show the overview

bsatarnejad committed:
writing a test for navigating from full view to details view of wp in boards module

bsatarnejad committed:
removing Trailing whitespaces

bsatarnejad committed:
removing Trailing whitespace

oliverguenther committed:
Ensure the keepTabService is loaded globally so it observes the change

If the keepTabService is not loaded globally, on boards it will first
load only at the work packages show view, incorrectly setting the
values.

oliverguenther committed:
Full view is only work-packages.show

oliverguenther committed:
Simplify details routes by checking with wildcard

oliverguenther committed:
Fix keep tab service spec
